cmd/tiller-proxy/internal/handler/handler.go

```diff
@@ -254,6 +254,16 @@ func (h *TillerProxy) ListReleases(w http.ResponseWriter, req *http.Request, par
 	response.NewDataResponse(apps).Write(w)
 }
 
+// TestRelease in the namespace given as Param
+func (h *TillerProxy) TestRelease(w http.ResponseWriter, req *http.Request, params Params) {
+	testResult, err := h.ProxyClient.TestRelease(params["releaseName"], params["namespace"])
```

Comment:
Before we run this we need to check if the user has permissions to create pods in the current namespace. You can check other methods to see how that can be done. The particularity of this method is that you don't need to resolve a manifest, you can just specify a simple manifest (because helm tests only create pods). I have not tested this code but it can be something like:

```
		userAuth := req.Context().Value(userKey).(auth.Checker)
                // helm tests only create pods so we only need to check that
                manifest := "apiVersion: v1\nkind: Pod"
		forbiddenActions, err := userAuth.GetForbiddenActions(params["namespace"], "create", manifest)
		if err != nil {
			response.NewErrorResponse(errorCode(err), err.Error()).Write(w)
			return
		}
		if len(forbiddenActions) > 0 {
			returnForbiddenActions(forbiddenActions, w)
			return
		}
```